test/test_train_mp_mnist.py

```diff
@@ -156,7 +158,10 @@ def test_loop_fn(loader):
     accuracy = xm.mesh_reduce('test_accuracy', accuracy, np.mean)
     return accuracy
 
-  train_device_loader = pl.MpDeviceLoader(train_loader, device)
+  # For profiling we're moving the mark step out of iterator
+  # i.e., batches_per_execution=-1
+  train_device_loader = pl.MpDeviceLoader(
+      train_loader, device, batches_per_execution=-1)
```

Comment:
discussed offline but noting here too:

adding kwargs like this `batches_per...` in order to get a clean profile is not user friendly. Let's figure out a way to automatically mark step at the end of the `StepTrace`, and not step anywhere else.

Reply:
Yeah seems a bit trickier than it seems as the step marking is embedded within the loader (which I'm not sure about whether that's the first place for it to be in the first place) + multithreading.

Reply:
what's the worry about multithreading?

Reply:
Updated the API, which now shouldn't require any changes from the user in terms of moving where the step is being marked.

Reply:
will look, thanks.